matscipy/dislocation.py

```diff
@@ -2704,10 +2570,34 @@ def __init__(self, left_dislocation, right_dislocation, burgers):
             ValueError: If one of the properties of
                         left and righ dislocations are not the same.
         """
+
+        self.left_dislocation = self.left_dislocation(a, C11, C12, C44, symbol)
+        self.right_dislocation = self.right_dislocation(a, C11, C12, C44, symbol)
+
+        # Change disloc burgers vectors, if requested
+        if self.new_left_burgers is not None:
+            self.left_dislocation.burgers_dimensionless = self.new_left_burgers.copy()
+        if self.new_right_burgers is not None:
+            self.right_dislocation.burgers_dimensionless = self.new_right_burgers.copy()
+        
+        # Set self.attrs based on left disloc attrs
+        left_dislocation = self.left_dislocation
+        right_dislocation = self.right_dislocation
+        self.crystalstructure = left_dislocation.crystalstructure
+        self.axes = left_dislocation.axes
+        self.unit_cell_core_position_dimensionless = left_dislocation.unit_cell_core_position_dimensionless
```

Comment:
check why some attributes come from left and some from right - do we need to check they are consistent?

Reply:
This was reproducing the old init, see the following code (currently line 2755 in dislocations.py):

super().__init__(left_dislocation.unit_cell,
                         left_dislocation.alat,
                         left_dislocation.C11,
                         left_dislocation.C12,
                         left_dislocation.C44,
                         left_dislocation.axes,
                         burgers,
                         unit_cell_core_position=left_dislocation.unit_cell_core_position,
                         parity=left_dislocation.parity,
                         glide_distance=right_dislocation.glide_distance,
                         n_planes=left_dislocation.n_planes,
                         self_consistent=left_dislocation.self_consistent)

Should be safe to change to left disloc glide distance, as there is a check that both are the same.